pinot-controller/src/main/java/org/apache/pinot/controller/util/ServerSegmentMetadataReader.java

```diff
@@ -264,6 +264,12 @@ public List<ValidDocIdsMetadataInfo> getValidDocIdsMetadataFromServer(String tab
         LOGGER.error("Unable to parse server {} response due to an error: ", streamResponse.getKey(), e);
       }
     }
+
+    Map<String, ValidDocIdsMetadataInfo> dedupedValidDocIdsMapping = new HashMap<>();
```

Comment:
Instead of adding this, we can change `validDocIdsMetadataInfos` to Map<String, ValidDocIdsMetadataInfo>

Reply:
Sure let me do that!